core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -570,162 +689,129 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId, boo
-            blockInfo.getFirst(), blockInfo.getSecond());
-      } catch (InvalidWorkerStateException e) {
-        // Evictor is not working properly
-        LOG.error("Failed to evict blockId {}, this is temp block", blockInfo.getFirst());
-        continue;
-      } catch (BlockDoesNotExistException e) {
-        LOG.info("Failed to evict blockId {}, it could be already deleted", blockInfo.getFirst());
-        continue;
-      }
-      synchronized (mBlockStoreEventListeners) {
-        for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-          listener.onRemoveBlockByWorker(sessionId, blockInfo.getFirst());
+      // Check minAvailableBytes is satisfied.
+      if (!availableBytesFound) {
+        if (evictorView.getAvailableBytes(location) >= minAvailableBytes) {
+          availableBytesFound = true;
         }
       }
-    }
-    // 2. transfer blocks among tiers.
-    // 2.1. group blocks move plan by the destination tier.
-    Map<String, Set<BlockTransferInfo>> blocksGroupedByDestTier = new HashMap<>();
-    for (BlockTransferInfo entry : plan.toMove()) {
-      String alias = entry.getDstLocation().tierAlias();
-      if (!blocksGroupedByDestTier.containsKey(alias)) {
-        blocksGroupedByDestTier.put(alias, new HashSet());
+
+      if (contigiousSpaceFound && availableBytesFound) {
+        break;
       }
-      blocksGroupedByDestTier.get(alias).add(entry);
-    }
-    // 2.2. move blocks in the order of their dst tiers, from bottom to top
-    for (int tierOrdinal = mStorageTierAssoc.size() - 1; tierOrdinal >= 0; --tierOrdinal) {
-      Set<BlockTransferInfo> toMove =
-          blocksGroupedByDestTier.get(mStorageTierAssoc.getAlias(tierOrdinal));
-      if (toMove == null) {
-        toMove = new HashSet<>();
+
+      if (!evictionCandidates.hasNext()) {
+        break;
       }
-      for (BlockTransferInfo entry : toMove) {
-        long blockId = entry.getBlockId();
-        BlockStoreLocation oldLocation = entry.getSrcLocation();
-        BlockStoreLocation newLocation = entry.getDstLocation();
-        MoveBlockResult moveResult;
+
+      long blockToDelete = evictionCandidates.next();
+      blocksIterated++;
+      if (evictorView.isBlockEvictable(blockToDelete)) {
         try {
-          moveResult = moveBlockInternal(Sessions.createInternalSessionId(),
-              blockId, oldLocation, newLocation);
-        } catch (InvalidWorkerStateException e) {
-          // Evictor is not working properly
-          LOG.error("Failed to demote blockId {}, this is temp block", blockId);
-          continue;
-        } catch (BlockAlreadyExistsException e) {
-          continue;
-        } catch (BlockDoesNotExistException e) {
-          LOG.info("Failed to demote blockId {}, it could be already deleted", blockId);
-          continue;
-        }
-        if (moveResult.getSuccess()) {
+          BlockMeta blockMeta = mMetaManager.getBlockMeta(blockToDelete);
+          removeBlockInternal(blockMeta);
```

Comment:
We don't need any locks for this?

Reply:
`freeSpaceInternal()` is only called by `allocateSpace()` which is required to be called with meta-write lock acquired.

We actually don't need this API in `BlockWorker` anymore.